pallets/amm-support/src/lib.rs

```diff
@@ -0,0 +1,170 @@
+// This file is part of hydration-node.
+
+// Copyright (C) 2020-2022  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::too_many_arguments)]
+
+use crate::types::*;
+use frame_support::sp_runtime::app_crypto::sp_core;
+use frame_support::sp_runtime::{BoundedVec, DispatchError, DispatchResult};
+use frame_system::pallet_prelude::BlockNumberFor;
+use sp_core::ConstU32;
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod tests;
+
+pub mod types;
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+const LOG_TARGET: &str = "runtime::amm-support";
+
+pub const MAX_STACK_SIZE: u32 = 10;
```

Comment:
What is the max size of the event at 10? Maybe we could increase to 16?

Reply:
bounded vec stored in storage somehow works only with u32, experienced this strange behaviour also in other pallets, not sure why...

Bumped to 16 anyway